inductiva/tasks/task.py

```diff
@@ -401,13 +410,66 @@ def download_outputs(
             warnings.warn("Path already exists, files may be overwritten.")
         output_dir.mkdir(parents=True, exist_ok=True)
 
+        download_message = "Downloading simulation outputs to %s"
+
+        if filenames and file_server_available:
+            partial_download_done = False
+
+            logging.info(download_message, output_dir)
+
+            try:
+                with remote_filesystem.open(download_url, "rb") as remote_file:
+                    with zipfile.ZipFile(remote_file) as remote_zip_file:
+                        for filename in filenames:
+                            try:
+                                remote_zip_file.extract(
+                                    filename,
+                                    path=output_dir,
+                                )
+                            except KeyError:
+                                logging.warning(
+                                    "File %s not found in the output archive.",
+                                    filename)
```

Comment:
this has a lot of cyclomatic complexity and a bit har to read. Can the code be refactored (maybe move the try block to a separate method)?

Reply:
Moved the block to `download_partial_outputs` function.